src/app/validators/validator.service.ts

```diff
@@ -38,25 +38,25 @@ constructor(private couchService: CouchService) {}
       })
     ).debug('Checking uniqueness of ' + fieldName + ' in ' + dbName);
   }
-
+// made validator function universal as oppoed to being specific to courses
   public isNameAvailible$(
     dbName: string,
     fieldName: string,
     ac: AbstractControl,
-    courseId: string
+    Id: string
   ): Observable<ValidationErrors | null> {
     return timer(500).pipe(
       switchMap(() => this.couchService.post(
         `${dbName}/_find`,
         findDocuments(
-          { 'courseTitle' : ac.value },
-          [ '_id', 'courseTitle' ]
+          { fieldName : ac.value },
```

Comment:
This object will have a key equal to "fieldName", not the value of the variable `fieldName`.  You need to surround fieldName in brackets (see [Computed property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names))